src/metax_api/services/pagination.py

```diff
@@ -1,58 +1,66 @@
 from rest_framework.pagination import LimitOffsetPagination
+# from django.db.models.query import QuerySet
 
 
 class DirectoryPagination(LimitOffsetPagination):
 
     page_size = 10
     page_size_query_param = 'page_size'
 
-    def paginate_queryset(self, queryset, request, view=None):
-        self.count = self.get_count(queryset)
-        self.limit = self.get_limit(request)
+    def paginate_queryset(self, dirs, files, request, view=None):
```

Comment:
I'm wondering if this method (also the get_count) should be renamed to reflect better that they are related to list and dict objects rather than querysets? By doing this, one would make the methods to handle only one type of inputs and also preserve these inherited methods for possible later use for actual querysets (though, this is is not going to happen anytime soon 😁)

Reply:
paginate_queryse actualy takes querysets and  lists and outputs list tuple. Comment added and changed name of paginate_queryset function. I don't think get_count should be changed as it suppose to be able to count all data structures.